lilo.c:219: warning: array subscript is below array bounds
common.c:175: warning: array subscript is below array bounds
bsect.c:1597: warning: array subscript is below array bounds

---
 bsect.c  |    3 ---
 common.c |    2 --
 lilo.c   |    6 ------
 3 files changed, 11 deletions(-)

--- a/src/bsect.c
+++ b/src/bsect.c
@@ -1594,9 +1594,6 @@ void bsect_uninstall(char *boot_dev,char
     if (*(unsigned short *) &bsect.sector[BOOT_SIG_OFFSET] != BOOT_SIGNATURE)
 	die("Boot sector of %s does not have a boot signature",boot_dev ?
 	  boot_dev : dev.name);
-    if (!strncmp(bsect.par_1.signature-4,"LILO",4))
-	die("Boot sector of %s has a pre-21 LILO signature",boot_dev ?
-	  boot_dev : dev.name);
     if (strncmp(bsect.par_1.signature,"LILO",4))
 	die("Boot sector of %s doesn't have a LILO signature",boot_dev ?
 	  boot_dev : dev.name);
--- a/src/common.c
+++ b/src/common.c
@@ -172,8 +172,6 @@ void check_version(BOOT_SECTOR *sect,int
 {
     int bs_major, bs_minor;
 
-    if (!strncmp(sect->par_1.signature-4,"LILO",4))
-	die("%s has a pre-21 LILO signature",name(stage));
     if (strncmp(sect->par_1.signature,"LILO",4))
 	die("%s doesn't have a valid LILO signature",name(stage));
     if ((sect->par_1.stage&0xFF) != stage)
--- a/src/lilo.c
+++ b/src/lilo.c
@@ -216,12 +216,6 @@ static void show_other(int fd)
 
     if (read(fd,buf,sizeof(buf)) != sizeof(buf))
 	die("Read on map file failed (access conflict ?) 1");
-    if (!strncmp(buf[0].par_c.signature-4,"LILO",4)) {
-	printf("    Pre-21 signature (0x%02x,0x%02x,0x%02x,0x%02x)\n",
-	  buf[0].par_c.signature[0],buf[0].par_c.signature[1],
-	  buf[0].par_c.signature[2],buf[0].par_c.signature[3]);
-	return;
-    }
     if (strncmp(buf[0].par_c.signature,"LILO",4)) {
 	printf("    Bad signature (0x%02x,0x%02x,0x%02x,0x%02x)\n",
 	  buf[0].par_c.signature[0],buf[0].par_c.signature[1],
